OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -36,7 +36,8 @@ public class BuildingInfo : ITraitInfo, IOccupySpaceInfo, IPlaceBuildingDecorati
 		[Desc("Where you are allowed to place the building (Water, Clear, ...)")]
 		public readonly HashSet<string> TerrainTypes = new HashSet<string>();
 
-		[Desc("The range to the next building it can be constructed. Set it higher for walls.")]
+		[Desc("The range to the next building it can be constructed. Set it higher for walls.",
+			"Set to zero to disable adjacency.")]
```

Comment:
Maybe -1 instead? 0 could also mean to not give adjacency. What you mean is disable adjacency checks though.